eth/txpool.go

```diff
@@ -13,24 +13,23 @@ import (
 )
 
 func (c *Client) txpoolTransaction(txHash string) {
-	// rawTX, err := rpc.EthGetTransactionByHash(txHash)
 	rawTx, err := c.Rpc.EthGetTransactionByHash(txHash)
 	if err != nil {
 		log.Errorf("c.Rpc.EthGetTransactionByHash:Get TX Err: %s", err.Error())
 	}
 	c.parseETHTransaction(*rawTx, -1, false)
 
 	c.parseETHMultisig(*rawTx, -1, false)
-	// log.Debugf("new txpool tx %v", rawTx.Hash)
 
 	// add txpool record
-	c.AddToMempoolStream <- pb.MempoolRecord{
-		Category: int32(rawTx.Gas),
-		HashTX:   rawTx.Hash,
+	if rawTx.GasPrice.IsInt64() {
+		c.AddToMempoolStream <- pb.MempoolRecord{
+			Category: rawTx.GasPrice.Int64(),
+			HashTX:   rawTx.Hash,
+		}
 	}
-	if strings.ToLower(rawTx.To) == strings.ToLower(c.Multisig.FactoryAddress) {
 
-		// go func() {
+	if strings.ToLower(rawTx.To) == strings.ToLower(c.Multisig.FactoryAddress) {
```

Comment:
let's leave goroutine launch inside IF statement, it's better to be non-blocking on IO operations